config/core/configmaps/features.yaml

```diff
@@ -132,6 +132,12 @@ data:
     # 2. Disabled: http2 connection will only be attempted when port name is set to "h2c".
     autodetect-http2: "disabled"
 
+    # Controls whether pod ip is injected using kubernetes downward API of statusp.podIP.
+    # This isn't available in AKS (and other?) virtual nodes: https://docs.microsoft.com/en-us/answers/questions/352186/pods-stuck-in-pending-state-on-virtual-nodes.html
+    # 1. Enabled: SERVING_POD_IP is injected into knative queue container.
+    # 2. Disabled: queue container attempts to determine IP dynamically
+    inject-pod-ip: "enabled"
```

Comment:
> I do wish AKS would create the pod with an empty string rather than keeping it pending forever though.. is there anything on the horizon that might allow that?

I haven't dug into the guts of how virtual kubelet works, but there has been some discussion about this: https://github.com/virtual-kubelet/virtual-kubelet/issues/123.  The short summary is that it doesn't seem settled on what the virtual kubelete base code is responsible for, and what each specific provider is responsible for.

I do like the idea of a more generic feature flag.  virtual-node-compatibility sounds perfect.  But given the above, if it does end up that Azure AKS doesn't support PodIP injection, but maybe AWS ECS does, what happens with that flag then?

I'm fine with whatever you all would like so just let me know.

Links to different virtual kubelet guides that have some more details:

[General serverless feature comparison](https://thenewstack.io/comparison-aws-fargate-vs-google-cloud-run-vs-azure-container-instances/)
[AWS](https://aws.amazon.com/blogs/opensource/aws-fargate-virtual-kubelet/)
[Azure](https://docs.microsoft.com/en-us/azure/aks/virtual-nodes)

Reply:
ok, very very sorry for slow response, this got lost in my notifications, but! I think I may have a reasonably nice solution to suggest here (hopefully, anyway). It looks like the only place we actually use the podIP is [in the OpenCensus tracer](https://github.com/knative/serving/blob/13ee889bb7b57d697d2c3054b90346a1f96acf4e/cmd/queue/main.go#L339), so what if we don't bother passing the PodIP [if tracingbackend == "none"](https://github.com/knative/serving/blob/13ee889bb7b57d697d2c3054b90346a1f96acf4e/cmd/queue/main.go#L335-L337) (since it won't be needed in that case, anyway). That would avoid needing any workarounds to get the pod ip locally, _and_ we won't need to figure out a good feature flag since we already have the tracingconfig in resources/queue.go. 

Obviously the downside is you wouldn't get tracing with virtualkubelet, but I'd argue maybe getting a fix upstream to at least let pods that use the downward api get created, even with an empty podIP, might be the best route to enabling that. Does that sound ok or would you be very sad if tracing did not work on virtual Kubelet (if so we'll need to figure out another plan, but even in that case this may still be a nice and easy first step to land this).